### DISCUSSION TITLE
Watch interrupted during high performance

### DISCUSSION THREAD

**Discussion by HermansG** [akkadotnet/akka.net#4735] [Q&A] [answered]
We have an Actor that calculates results in approx. 10ms.
The calculation was started with .NET's `Task.StartNew` combined with `PipeTo(Self)`.
The FrontEnd has a Watch with the parent of the Actor.
We encountered situations with high perfomance > 1000 calculations in a row where the Watch failed at the FrontEnd.
We upgraded 1.4.12 to 1.4.14 with the same result.
We changed the `Task.StartNew` and used child actors instead.
Our idea was that it would be better to use resources inside of the Akka system instead of outside with Windows `Task`'s.
Observation was that the calculation was a fraction slower but the stability improved. But still the errors could occur.
We then added `watch-failure-detector` to the Akka configuration.
threshold = 100.0, acceptable-heartbeat-pause = 20 s
This improved the behaviour enough.
Still it is theoretically possible that under high pressure (high CPU and memory) the Watch can fail.
What do you think about this our try to improve stability?
Are there other improvements that you can think of?

**Comment by Aaronontheweb** [comment, marked as answer]
@HermansG does the watch fail because it never gets sent across the wire? Or because the watching actor never receives a `Terminated`? 

(I know it might be tough to gather that information, but it would help us understand where the source of the problem might be)

**Reply by Aaronontheweb** [reply]
Hi @HermansG - did you see my previous message?

**Reply by HermansG** [reply]
I saw it, I was not able to find the answer. Currently the system is stable again, I am not experienced enough to tackle the underlying mechanisms.

**Reply by Aaronontheweb** [reply]
You might have been affected by this issue: https://github.com/akkadotnet/akka.net/issues/4849 

Those fixes will be included as part of [Akka.NET v1.4.19](https://github.com/akkadotnet/akka.net/milestone/49).

**Reply by Aaronontheweb** [reply]
Was this issue resolved for you eventually @HermansG ?

**Reply by HermansG** [reply]
Indeed resolved: no problems currently. It was difficult to understand then (now almost a year ago).

**Reply by Aaronontheweb** [reply]
Some of these issues with the protocol can be difficult to track down but I'm glad this is resolved.
